### DISCUSSION TITLE
Outputting custom HTML from a custom directive

### DISCUSSION THREAD

**Discussion by noelwelsh** [planet42/Laika#316] [Q&A] [answered]
I'm attempting to create a directive that outputs custom HTML, but I can't get it to work.  (This will only work with HTML output, but that's fine for my use case. ) My use case is to include custom snippets of Javascript so I can embed animations and interactive demonstrations in my documentation. (I need to call functions to kick off selected animations for each page, as there will be hundreds across all my documentation.) I've tried many things but I can't get Laika to not quote or otherwise mess with my HTML in some way. I just want it to appear verbatim in the output. Is this possible?

**Comment by jenshalm** [comment, marked as answer]
Yes, the AST node for passthrough content is `RawContent`, for example:

```scala
RawContent(NonEmptySet.one("html"), "<p>Hello</p>")
```

The node will simply be ignored for all output formats not contained in the format set.

**Reply by noelwelsh** [reply]
Thanks!
